content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md

```diff
@@ -15,17 +15,34 @@ related:
   - /influxdb/cloud-serverless/reference/client-libraries/v3/
 ---
 
-## Troubleshoot query performance
-
-Use the following tools to help you identify performance bottlenecks and troubleshoot problems in queries:
+Optimize your queries to reduce their memory and compute (CPU) requirements.
+Use tools to help you identify performance bottlenecks and troubleshoot problems in queries.
 
 <!-- TOC -->
 
+- [Strategies for improving query performance](#strategies-for-improving-query-performance)
 - [Troubleshoot query performance](#troubleshoot-query-performance)
   - [EXPLAIN and ANALYZE](#explain-and-analyze)
   - [Enable trace logging](#enable-trace-logging)
 
-<!-- /TOC -->
+
+## Strategies for improving query performance
+
+A query may be slow due to the following reasons:
+
+- It queries a large time-range of data.
+- It includes intensive operations, such as `ORDER BY`.
+- The query plan isn't optimal--for example, applying the same sort (`ORDER BY`) to already sorted data.
+- It needs to retrieve many parquet files from object storage. The same query performs better if it retrieves fewer - though, larger - files.
+- It queries many overlapped parquet files.
+- It queries many string values. A query against a field that stores integers outperforms a query against string data.
+
+Follow these strategies to help improve query performance:
+
+- Follow [schema design best practices](/influxdb/serverless/write-data/best-practices/schema-design/) to make your data easier to query.
+- [Downsample data](/influxdb/serverless/process-data/downsample/).
+- (If you need custom-partitioning for your data, contact Sales/Support for [CST/Clustered]?)
+- If resource usage and costs allow, “prewarm” query caches by running the query a few times.
```

Comment:
I don't think it is feasible to "prewarm" on serverless

Reply:
I also don't think it is useful for a user to pre-warm a cache. That is like "let me wait for a slow query so that my next query can be fast". Pre-warming is something the system has to do to attempt to guess the next query and typically this is good in systems where there are relationships between data sets (if you query A then likely they are going to want B next due to entity relationship of some form) or where the system can learn typical user behaviour and attempt to get ahead of them.

Yes it is possible for an application layer above Influx to do that but I am not of a mind to instruct our users to fill gaps that Influx should be filling. We could propose it but we should not rest on it ourselves and think we can avoid taking ownership of that problem ourselves.